src/sources/http_server.rs

```diff
@@ -333,20 +334,20 @@ impl SourceConfig for SimpleHttpConfig {
         )
     }
 
-    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<SourceOutput> {
         // There is a global and per-source `log_namespace` config.
         // The source config overrides the global setting and is merged here.
         let log_namespace = global_log_namespace.merge(self.log_namespace);
 
         let schema_definition = self.schema_definition(log_namespace);
 
-        vec![Output::default(
+        vec![SourceOutput::new_logs(
             self.decoding
                 .as_ref()
                 .map(|d| d.output_type())
                 .unwrap_or(DataType::Log),
-        )
-        .with_schema_definition(schema_definition)]
+            schema_definition,
+        )]
```

Comment:
What happens here if the decoder is `native` or `native_json` and can product metrics? Would that need a `SourceOutput::new_metrics()`?

Reply:
It's fine. It's only `new_logs` because it needs a schema definition and logs need a schema definition. If we pass in metrics with a schema definition it won't break things.

Reply:
It feels a little awkward to me, but if it's not breaking things I've got no real objection here.